### PR TITLE
Animate your visualizations guide

### PR DESCRIPTION
**Pull request by elenatorro**
Related to #974 

* Guide at staging:
http://carto-vl-974-animate-your-visualizations-guide.developers.carto-staging.com/developers/carto-vl/guides/animate-your-visualizations/

### CODE REVIEW

**Comment by makella** [comment]
@elenatorro i took a look through, and have a few comments... i'm not sure what the best way to comment is.

I started re-writing but that probably isn't the best thing to do.

I think that it is great that you added the control parts, but I think the flow from the beginning is hard to follow.

I know it is difficult to restructure an existing guide!!

anyway, i'll paste what you have into a google doc, add some comments, and maybe we can go from there.

**Comment by makella** [comment]
@elenatorro I made a comment where I left off... I will continue reviewing tomorrow.

Here is the google doc:
https://docs.google.com/document/d/1jmo6r9EF1LPIU6n_hwpZze2Uh0Zijdmuzm0IzjbqrUM/edit

feel free to ping me if you want to talk about any of it.

thanks!
mamata

File: docs/guides/0x-animate-your-visualizations.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+Before we begin, it is important to understand the general syntax for animating data in CARTO VL:
+
+`filter: animation(input, duration, fade(fadeIn, fadeOut))`
+
+
+We will walk through each part in more detail throughout this guide.
+
+Using the syntax above, let’s create a basic animation where points appear and disappear on a map. 
+
+We will use the attribute `date_time` (a record of time when a data point was collected) as our input, set the duration of the animation to `10` seconds and set both fade parameters to `1`.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: It would be good to first describe which dataset we are going to use

File: docs/guides/0x-animate-your-visualizations.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+```js
+const viz = new carto.Viz(`
+    filter: animation($date_time, 10, fade(1,1))
+`);
+```
+
+<div class="example-map">
+    <iframe
+        id="animation-step-0"
+        src="/developers/carto-vl/examples/maps/guides/animation/step-0.html"
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: would it be nice to start also this guide with step-1 ?

File: docs/guides/0x-animate-your-visualizations.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+        width="100%"
+        height="500"a
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### The filter property
+
+
+Now that we have a basic map of animated points, let’s get a deeper understanding of the different pieces that make up the animation expression to fine-tune the visualization.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Aren't we missing some code for the user to start working in this guide? I mean how is the layer configured and added to the map?

File: docs/guides/0x-animate-your-visualizations.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+const viz = new carto.Viz(`
+    filter: animation($date_time, 10, fade(1,1))
+`);
+```
+
+The result is an animation that cycles through all of the records in the data over 10 seconds. Each point will take 1 second to become visible (fade in) and another second to disappear (fade out).
+
+
+```js
+const viz = new carto.Viz(`
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Here it seems that we are repeating the same contents as in the previous code!

File: docs/guides/0x-animate-your-visualizations.md (hunk 3 above)
Comment (VictorVelarde) [reply docs/guides/0x-animate-your-visualizations.md]: Maybe we could skip the basics (template / const map = ...) but I would add the source and layer creation in a snippet. And then, point the user to the working html

File: docs/guides/0x-animate-your-visualizations.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### The filter property
+
+
+Now that we have a basic map of animated points, let’s get a deeper understanding of the different pieces that make up the animation expression to fine-tune the visualization.
+
+In the example above, the animation expression is set to the `filter` property (for more information see the[vizSpec](https://carto.com/developers/carto-vl/reference/#vizspec)). The convention in CARTO VL is that `0` represents the boolean value `false`, (the absence of or an *off* state), while `1` represents `true` (the presence of or an *on* state). In the context of an  animation expression assigned to a `filter`, any feature that has a  value of `0` will be “filtered-out” and not shown while any feature that meets the filter, is assigned a value of `1` and is shown.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: add an space between the_[vizSpec]

File: src/renderer/viz/expressions.js [hunk 6]
```diff
@@ -6,7 +6,7 @@
  *  - **strokeColor**: stroke/border color of points and polygons, not applicable to lines
  *  - **width**: fill diameter of points, thickness of lines, not applicable to polygons
  *  - **strokeWidth**: stroke width of points and polygons, not applicable to lines
- *  - **filter**: filter features by removing from rendering and interactivity all the features that don't pass the test
+ *  - **filter**: filter features by removing from rendering and interactivity all the features that don't pass the expression's result.
```
Comment (davidmanzanares) [inline src/renderer/viz/expressions.js]: pass an "expression's result."? sounds wrong to me

File: src/Viz.js [hunk 7]
```diff
@@ -78,7 +78,7 @@ const SUPPORTED_PROPERTIES = [
 * @property {Number} width - fill diameter of points, thickness of lines, not applicable to polygons
 * @property {Color} strokeColor - stroke/border color of points and polygons, not applicable to lines
 * @property {Number} strokeWidth - stroke width of points and polygons, not applicable to lines
-* @property {Number} filter - filter features by removing from rendering and interactivity all the features that don't pass the test. In combination with {@link carto.expressions.animation} temporal maps can be created.
+* @property {Number} filter - filter features by removing from rendering and interactivity all the features that don't pass the expression's result. In combination with {@link carto.expressions.animation} temporal maps can be created.
```
Comment (davidmanzanares) [inline src/Viz.js]: same

File: docs/guides/0x-animate-your-visualizations.md (hunk 5 above)
Comment (VictorVelarde) [reply docs/guides/0x-animate-your-visualizations.md]: The url should point to '/developers/carto-vl/reference' and not the absolute url at carto.com

Comment (VictorVelarde) [reply docs/guides/0x-animate-your-visualizations.md]: I personally find the `(the absence of or an off state)` phrase (and its contrary) a little confusing. It may be enough saying that `0 represents false`

File: docs/guides/0x-animate-your-visualizations.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+</div>
+
+#### The filter property
+
+
+Now that we have a basic map of animated points, let’s get a deeper understanding of the different pieces that make up the animation expression to fine-tune the visualization.
+
+In the example above, the animation expression is set to the `filter` property (for more information see the[vizSpec](https://carto.com/developers/carto-vl/reference/#vizspec)). The convention in CARTO VL is that `0` represents the boolean value `false`, (the absence of or an *off* state), while `1` represents `true` (the presence of or an *on* state). In the context of an  animation expression assigned to a `filter`, any feature that has a  value of `0` will be “filtered-out” and not shown while any feature that meets the filter, is assigned a value of `1` and is shown. 
+
+You can think of the  animation expression like a clock, cycling through each record of the input.  Any time there is a match, ,  the feature is drawn.  Now that we’ve gone through a more in-depth discussion of how features draw, let’s take a look at the next set of animation parameters are used to define the property, speed, and transition between animated features.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: "of how features are drawn"?

File: docs/guides/0x-animate-your-visualizations.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+
+#### Input
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Maybe it's better to just use bulleted points for these 3 items, or even a table. I find it would be a clearer way of displaying them

File: docs/guides/0x-animate-your-visualizations.md [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+
+The first animation parameter (input), is the attribute that you want to animate. By default, the animation progresses from the attribute minimum to maximum values.
+#### Duration
+
+This parameter defines the duration of the animation progress in seconds. As stated above, during the animation cycle, all possible input values are evaluated. When an input value is matched, animation returns a value of `1` for the feature and it is drawn.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: How about removing 'progress' from this sentence?

File: docs/guides/0x-animate-your-visualizations.md [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+
+#### Fade
+
+The `fade` parameter is used to define two additional (`fadeIn, fadeOut`) durations in seconds. These parameters allow for smooth transitions between features during an animation's duration. During the *`fadeIn`* phase, all features with a match will fade-in to the animation, transitioning from `0` (invisible) to `1` (visible). During the *`fadeOut`* phase features will transition from `1` back to `0` and the next set of features will begin to fade-in.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: "during the animation" instead of "during an animation's duration"?

File: docs/guides/0x-animate-your-visualizations.md [hunk 12]
```diff
@@ -0,0 +1,391 @@
...
+
+### Style and configure the animation
+
+To illustrate these concepts, let's animate the journey of three birds from January to April of 2014 using data from [movebank.org](https://www.movebank.org/).
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: This is what I was missing at the beginning... when presenting the viz

File: docs/guides/0x-animate-your-visualizations.md [hunk 13]
```diff
@@ -0,0 +1,391 @@
...
+
+#### Animate points
+
+At the beginning of the guide we created a simple point animation using the attribute `$date_time` for the temporal input (the date of the bird track and associated timestamp), set the duration to `10` seconds and both fade parameters to `1`.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: I feel like this and the first "Create a basic animation" should be revised (maybe even merged somehow), and include more details (and code snippets) for the user to follow the guide in a "hands-on mode"

File: docs/guides/0x-animate-your-visualizations.md [hunk 14]
```diff
@@ -0,0 +1,391 @@
...
+        src="/developers/carto-vl/examples/maps/guides/animation/step-1.html"
+        width="100%"
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### Adjust the duration
+
+In the animation above, we can see that the birds spend the majority of this three month period in West Africa and then eventually migrate to Northern Europe. To see the journey in more detail, we'll adjust the duration to `30` seconds. Now looking at the map, we can see that each bird, at a different time, flies north along the coast of Morocco to Eastern Europe.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: ✨ Nice dataset and story!

File: docs/guides/0x-animate-your-visualizations.md [hunk 15]
```diff
@@ -0,0 +1,391 @@
...
+        frameBorder="0">
+    </iframe>
+</div>
+
+#### Final touches
+
+Since this visualization represents the journey of three different birds, we can assign  a unique color to  each one  using their names to define `buckets` inside of a `ramp`.decrease the `width` of the points to `4` and finally, remove the `strokeWidth` by setting it to `0`. 
+
+*Note:*
+For a more in-depth discussion of ramps and other styling properties, check out our [Data Driven Visualizations Guide](/developers/carto-vl/guides/0x-data-driven-visualizations.md).
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Great!. We'll have to check this url later on, when https://github.com/CartoDB/carto-vl/pull/1005 will be integrated

File: docs/guides/0x-animate-your-visualizations.md [hunk 16]
```diff
@@ -0,0 +1,391 @@
...
+    color: ramp(buckets($bird_name, ["Sanne", "Eric", "Nico"]),[deeppink, yellow, turquoise])
+    width: 4
+    strokeWidth: 0
+`);
+```
+
+<div class="example-map">
+    <iframe
+        id="animation-step-4"
+        src="/developers/carto-vl/examples/maps/guides/animation/step-4.html"
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: 👀 It looks like this is failing because of a little syntax error!

File: docs/guides/0x-animate-your-visualizations.md [hunk 17]
```diff
@@ -0,0 +1,391 @@
...
+        height="500"
+        frameBorder="0">
+    </iframe>
+</div>
+
+### Adding controls to your animation
+
+Congratulations! You have created an animated visualization. 
+
+Next, let’s add some animation controls to play and pause the animation as well as a slider to adjust the speed of the animation on-the-fly.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Great!

File: docs/guides/0x-animate-your-visualizations.md [hunk 18]
```diff
@@ -0,0 +1,391 @@
...
+Continuing with our map of bird migration, we will add  two variables in our Viz object: `@duration` and `@animation`. The `@duration` variable defines how to modify the duration value, which is really useful for playing with the animation expression.
+```js
+const viz = new carto.Viz(`
+    @duration: 30
+    @animation: animation($date_time, @duration, fade(0, 0.5))
+    filter: @animation
+`);
+```
+
+The different variables are available in the `viz` object you have just created. If you want to access the animation variable you can do it by typing `viz.variables.animation`. By assigning the animation expression to the `@animation` variable, you are going to be able to call the [public methods](http://animation-guide.developers.carto-staging.com/developers/carto-vl/reference/#expressionsanimation) as follows:
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Let's change that url to just `/developers/whatever...`

File: docs/guides/0x-animate-your-visualizations.md [hunk 19]
```diff
@@ -0,0 +1,391 @@
...
+
+The   elements we will add are:
+
+* **Progress** slider to display the animation progress.
+* **Play** button to play or resume the animation.
+* **Pause** button to pause the animation.
+* **Duration** slider to control the speed of the animation.
+
+```html
+<aside>
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: It would be nice to tell the reader something introducing this snippet. "To show the controls, we're going to create a panel. You have to add this just under the map div...."...

File: docs/guides/0x-animate-your-visualizations.md [hunk 20]
```diff
@@ -0,0 +1,391 @@
...
+$playButton.addEventListener('click', () => {
+    viz.variables.animation.play();
+});
+```
+
+In the same way, we can  update the **duration** of the animation when the duration range is adjusted in the slider:
+
+```js
+$durationRange.addEventListener('change', () => {
+    viz.variables.duration = parseInt($durationRange.value, 10);
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: I really like this part of the guide!

File: docs/guides/0x-animate-your-visualizations.md [hunk 21]
```diff
@@ -0,0 +1,391 @@
...
+
+```js
+setInterval(updateProgress, 100);
+```
+
+> Note: You can also do this by listening to layer events, you will learn how to use these events in the [Interactivity and Events Guide](09-interactivity-events.md)
+
+#### All together
+
+Well done! In the example below you will find the full code, including with the animation controls. You can play with it and make adjustments.
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Remove "with"?

File: docs/guides/0x-animate-your-visualizations.md [hunk 22]
```diff
@@ -0,0 +1,391 @@
...
+
+```js
+$durationRange.addEventListener('change', () => {
+    viz.variables.duration = parseInt($durationRange.value, 10);
+});
+```
+
+#### Update the progress
+
+You have learned how change the animation by interacting with the different buttons, and now you are going learn how to update the current progress of the animation. This function will be the one responsible of updating the progress range by updating the range value with the result of `getProgressPct()` method:
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: How to change?

File: docs/guides/0x-animate-your-visualizations.md [hunk 23]
```diff
@@ -0,0 +1,391 @@
...
+```js
+function updateProgress () {
+    $progressRange.value = viz.variables.animation.getProgressPct();
+}
+```
+
+Let’s call this function periodically by using `setInterval`:
+
+```js
+setInterval(updateProgress, 100);
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: The controls are great, but I miss here the **Date** itself!, maybe displayed in a simple label. 
I miss it so badly... 😄

File: docs/guides/0x-animate-your-visualizations.md [hunk 24]
```diff
@@ -0,0 +1,391 @@
...
+```
+
+> Note: You can also do this by listening to layer events, you will learn how to use these events in the [Interactivity and Events Guide](09-interactivity-events.md)
+
+#### All together
+
+Well done! In the example below you will find the full code, including with the animation controls. You can play with it and make adjustments.
+
+Remember you can style your HTML controls using CSS. These styles are very simple, but at the beginning of this guide you have a complete example that includes beautiful styles. This example is also available in the [examples section](http://carto.com/developers/carto-vl/examples/#example-animation-controls).
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: /developers url

Comment (VictorVelarde) [reply docs/guides/0x-animate-your-visualizations.md]: I see we don't have here the **/map/** fragment anymore... Have we get ridden of that? 

If the answer is YES, then we should talk a bit about when do we update all the affected urls in our guides

File: docs/guides/0x-animate-your-visualizations.md [hunk 25]
```diff
@@ -0,0 +1,391 @@
...
+    $progressRange.value = viz.variables.animation.getProgressPct();
+}
+
+setInterval(updateProgress, 100);
+
+// Add the layer to the map
+layer.addTo(map);
+```
+
+### Advanced animations
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Maybe we can add a <hr> or an "EXTRA" word to remark this is a bonus-part of the guide!

File: docs/guides/0x-animate-your-visualizations.md [hunk 26]
```diff
@@ -0,0 +1,391 @@
...
+
+In this section, you will learn other properties that can be used to animate the visualization. You can change the input range of values by using a `linear` expression, and thus visualize a subset of the data. For example, if a dataset spans an entire year, but you only want to animate between the months of February and June, you can adjust the input parameter to:
+
+```js
+const viz = new carto.Viz(`
+    animation(linear($month, 2, 6))
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: I would explain somehow that this is thx to the dataset having a $month column. Cause somebody could think this is done automagically!

File: docs/guides/0x-animate-your-visualizations.md [hunk 27]
```diff
@@ -0,0 +1,391 @@
...
+`);
+```
+
+Similarly, if you have a timestamp property (`$date`), you can select a specific range using the [`time`](https://carto.com/developers/carto-vl/reference/#cartoexpressionstime) expression:
+
+```js
+const viz = new carto.Viz(`
+    animation(linear($date_time, time('2018-01-01T00:00:00'), time('2018-01-05T00:00:00'))
```
Comment (VictorVelarde) [inline docs/guides/0x-animate-your-visualizations.md]: Maybe we could say here we are using ISO time format?

File: examples/guides/animation/step-5.html [hunk 28]
```diff
@@ -0,0 +1,256 @@
...
+        </section>
+    </aside>
+
+    <script>
+        // Add basemap and set properties
+        const map = new mapboxgl.Map({
+            container: 'map',
+            style: carto.basemaps.darkmatter,
+            center: [-22.6995, 41.94674],
+            center: [-22.6995, 41.94674],
```
Comment (VictorVelarde) [inline examples/guides/animation/step-5.html]: 👀 center attribute is duplicated here!

File: examples/guides/animation/step-6.html [hunk 29]
```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <!-- Include CARTO VL JS -->
+    <script src="../../../../dist/carto-vl.js"></script>
```
Comment (VictorVelarde) [inline examples/guides/animation/step-6.html]: I think all script references need one level less!, like --> **../../..**

File: examples/guides/animation/step-6.html [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+    <script>
+        // Add basemap and set properties
+        const map = new mapboxgl.Map({
+            container: 'map',
+            style: carto.basemaps.darkmatter,
+            center: [-22.6995, 41.94674],
+            zoom: 2.8,
+            scrollZoom: false
+        });
+        map.touchZoomRotate.disableRotation();
```
Comment (VictorVelarde) [inline examples/guides/animation/step-6.html]: Now that we have tilt and rotation, we could remove this from these examples, so animations can be displayed also in 2.5D

File: examples/interactivity/animation-controls.html [hunk 31]
```diff
@@ -35,13 +35,8 @@
         }
```
Comment (VictorVelarde) [inline examples/interactivity/animation-controls.html]: Do we need this whole file? I think we can safely remove it!

File: src/Viz.js [hunk 32]
```diff
@@ -78,7 +78,7 @@ const SUPPORTED_PROPERTIES = [
 * @property {Number} width - fill diameter of points, thickness of lines, not applicable to polygons
 * @property {Color} strokeColor - stroke/border color of points and polygons, not applicable to lines
 * @property {Number} strokeWidth - stroke width of points and polygons, not applicable to lines
-* @property {Number} filter - filter features by removing from rendering and interactivity all the features that don't pass the test. In combination with {@link carto.expressions.animation} temporal maps can be created.
```
Comment (VictorVelarde) [inline src/Viz.js]: Nice!

**Comment by makella** [comment]
@elenatorro @VictorVelarde 

I finished reviewing the text... more comments/suggestions here:
https://docs.google.com/document/d/1jmo6r9EF1LPIU6n_hwpZze2Uh0Zijdmuzm0IzjbqrUM/edit

File: docs/guides/0x-animate-your-visualizations.md (hunk 16 above)
Comment (makella) [reply docs/guides/0x-animate-your-visualizations.md]: ahhhh! good catch

**Comment by makella** [comment]
@VictorVelarde do you mind reviewing the text in the google doc? I don't want to add here because there are some questions I have in there.

I also finished going through the rest of the Guide, so it will be a more complete review at this point.

thanks!!

**Comment by VictorVelarde** [comment]
This is what I have been working on here:
- Mamata's CR. @makella I couldn't see any "comments" in the gdocs (maybe permissions?), but I've included its whole content in a commit here (and I could see some copy changes, here and there...). Now we can have just one place to review, this PR, if we need to make more changes.
- My CR. I have integrated my previous comments and suggestions in the guide, in two blocks:
    - basic improvements & fixes
    - more 'advanced' changes

File: docs/guides/0x-animate-your-visualizations.md (hunk 23 above)
Comment (VictorVelarde) [reply docs/guides/0x-animate-your-visualizations.md]: Added a currentTime label

**Comment by VictorVelarde** [comment]
After my meeting with @makella I have integrated her gdocs detailed feedback / changes into this PR, updating also the branch with all the recent changes in master

File: docs/guides/08-animate-your-visualizations.md [hunk 33]
```diff
@@ -0,0 +1,410 @@
...
+**Note:**
+The `setInterval()` is a commond method available at the browser, that calls a function or evaluates an expression at specified intervals, in milliseconds. You can also do this update by listening to `carto.layer` events. This topic is covered more in-depth in the [Interactivity and Events Guide](/developers/carto-vl/guides/interactivity-events/); once you know how, the second options is preferred.
+
+
+#### All together
+Well done! In the example below you will find the full code, including the animation controls. You can play with it and make adjustments.
+
+Remember you can style your HTML controls using CSS. These styles are very simple, but at the beginning of this guide you have a complete example that includes beautiful styles. This example is also available in the [examples section](/developers/carto-vl/examples/#example-animation-controls).
+
+<div class="example-map">
```
Comment (davidmanzanares) [inline docs/guides/08-animate-your-visualizations.md]: Clicking on the progress bar is broken

File: docs/guides/08-animate-your-visualizations.md [hunk 34]
```diff
@@ -0,0 +1,410 @@
...
+`);
+```
+
+You can also apply the same time range (`@time`) to other properties like color using a `ramp` expression. That double-encoding of the same time dimension, within the filter and the color, will reinforce the expressiveness of the visualization:
+
+```js
+const viz = new carto.Viz(`
+    @duration: 10
+    @animation: animation(@time, @duration, fade(0,0.5))
+    @time: linear($date_time, time('2014-03-30T20:24:25Z'), time('2014-04-24T23:52:14Z'))
```
Comment (davidmanzanares) [inline docs/guides/08-animate-your-visualizations.md]: I think `@time` is a bad name for this

File: docs/guides/08-animate-your-visualizations.md [hunk 35]
```diff
@@ -0,0 +1,410 @@
...
+        frameBorder="0">
+    </iframe>
+</div>
+
+
+#### Final touches
+Since this visualization represents the journey of three different birds, we can assign a unique color to each one, using their names to define `buckets` inside of a `ramp`. We will also decrease the `width` of the points to `4` and finally, remove the `strokeWidth` by setting it to `0`.
+
+**Note:**
+For a more in-depth discussion of ramps and other styling properties, check out our **Data-driven visualization guide** [part 1](/developers/carto-vl/guides/data-driven-visualizations-part-1/) and [part 2](/developers/carto-vl/guides/data-driven-visualizations-part-2/)
```
[the remark this answers is not in the document]
Comment (VictorVelarde) [reply docs/guides/08-animate-your-visualizations.md]: Thx!

File: docs/guides/08-animate-your-visualizations.md (hunk 33 above)
Comment (VictorVelarde) [reply docs/guides/08-animate-your-visualizations.md]: @davidmanzanares The progress bar in this example-5 has been kept "simple", not interactive, to avoid more code at this step. But we have that feature in example-7.

What I have found is a little bug, regarding to changing the animation's duration to 0 as it is running, I'll create the issue

Comment (VictorVelarde) [reply docs/guides/08-animate-your-visualizations.md]: As that one gets solved, I'm gonna set the minimum duration in the slider to 1

File: docs/guides/08-animate-your-visualizations.md (hunk 34 above)
Comment (VictorVelarde) [reply docs/guides/08-animate-your-visualizations.md]: @davidmanzanares any suggestion? how about @timeSteps?

**Comment by VictorVelarde** [comment]
I have just updated the [staging for this guide](http://carto-vl-974-animate-your-visualizations-guide.developers.carto-staging.com/developers/carto-vl/guides/animate-your-visualizations/).

@makella you can make a final review and, you're ok, we could merge this.
